agent/mapcall/src/main/java/randoop/instrument/MethodSignature.java

```diff
@@ -0,0 +1,277 @@
+package randoop.instrument;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Objects;
+import org.apache.bcel.generic.ConstantPoolGen;
+import org.apache.bcel.generic.InvokeInstruction;
+import org.apache.bcel.generic.Type;
+import plume.BCELUtil;
+import plume.UtilMDE;
+
+/**
+ * Defines a method in a way that can be used to substitute method calls using BCEL. A method is
+ * represented by its fully-qualified name and parameter types as BCEL {@code Type}.
+ *
+ * <p>Note: this is similar to the Randoop {@code randoop.reflection.RawSignature} class, but uses
+ * BCEL {@code Type} instead of {@code java.lang.reflect.Class} for the parameter types.
+ */
+public class MethodSignature {
+
+  /** fully-qualified class name */
+  private final String classname;
+
+  /** simple method name */
+  private final String name;
+
+  /** The parameter types */
+  private final Type[] paramTypes;
+
+  /** Cached {@link java.lang.reflect.Method} object for this {@link MethodSignature} */
+  private Method method;
+
+  /**
+   * Creates a {@code MethodSignature}.
+   *
+   * @param classname the fully-qualified classname
+   * @param name the method name
+   * @param argTypes the parameter types for the method
+   */
+  private MethodSignature(String classname, String name, Type[] argTypes) {
+    this.classname = classname;
+    this.name = name;
+    this.paramTypes = argTypes;
+    this.method = null;
+  }
+
+  /**
+   * Creates a {@link MethodSignature} object for a {@code java.lang.reflect.Method} object.
+   *
+   * @param method the reflection method object
+   * @return the {@link MethodSignature} representation of the reflection object
+   */
+  public static MethodSignature of(Method method) {
```

Comment:
All of the `of` methods in this class are conversions.  Let me know if you would prefer they be constructors.